ext/opentelemetry-ext-celery/src/opentelemetry/ext/celery/utils.py

```diff
@@ -0,0 +1,161 @@
+# Copyright The OpenTelemetry Authors
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+# Celery Context key
+CTX_KEY = "__otel_task_span"
+
+
+# pylint:disable=too-many-branches
+def set_attributes_from_context(span, context):
+    """Helper to extract meta values from a Celery Context"""
+
+    attribute_keys = (
+        "compression",
+        "correlation_id",
+        "countdown",
+        "delivery_info",
+        "declare",
+        "eta",
+        "exchange",
+        "expires",
+        "hostname",
+        "id",
+        "priority",
+        "queue",
+        "reply_to",
+        "retries",
+        "routing_key",
+        "serializer",
+        "timelimit",
+        "origin",
+        "state",
+    )
+
+    for key in attribute_keys:
+        value = context.get(key)
+
+        # Skip this key if it is not set
+        if value is None or value == "":
+            continue
+
+        # Skip `timelimit` if it is not set (it's default/unset value is a
+        # tuple or a list of `None` values
+        if key == "timelimit" and value in [(None, None), [None, None]]:
+            continue
+
+        # Skip `retries` if it's value is `0`
+        if key == "retries" and value == 0:
+            continue
+
+        # Celery 4.0 uses `origin` instead of `hostname`; this change preserves
+        # the same name for the tag despite Celery version
+        if key == "origin":
+            key = "hostname"
+
+        # TODO: hack to avoid bad attribute type
+        if key == "delivery_info":
+            # Get also destination from this
+            routing_key = value.get("routing_key")
+            if routing_key:
+                span.set_attribute("messaging.destination", routing_key)
+            value = str(value)
+
+        attribute_name = "celery.{}".format(key)
+
+        if key == "id":
```

Comment:
this block of code feels like it could be simplified (e.g. using a map for known constants, or a lookup).